index.bs

```diff
@@ -96,144 +96,149 @@ See also:
-did not match reality.
-
-Asking users for consent via permission prompts
-can reinforce the idea that <a href="#safe-to-browse">the web is safe</a>
-by showing the user that certain things won't happen without their permission.
-But frequently asking users for consent can also show how scary a place the web is
-by showing how many sites are willing to ask for intrusive and unnecessary permissions.
-
-<h3 id="devices-platforms">Support the full range of devices and platforms</h3>
+If a useful feature has the potential to cause harm to users,
+make sure that the user can give *meaningful consent* for that feature to be used,
+and that they can refuse consent effectively.
+
+In order to give *meaningful consent*, the user must:
+- **understand** what permission they may choose whether to grant the web page
+- be able to choose to give or refuse that permission **effectively**.
+
+If a feature is powerful enough to require a user consent,
+but it's impossible to explain to a typical user what they are consenting to,
+that's a signal that you may need to reconsider the design of the feature.
+
+If a permission prompt is shown,
+and the user doesn't grant permission,
+the Web page should not be able to do anything
+that the user believes they have refused consent for.
+
+By asking for consent,
+we can inform the user of what capabilities the web page does or doesn't have,
+reinforcing their confidence that <a href="#safe-to-browse">the web is safe</a>.
+However, the <a href="priority-of-constituencies">user benefit</a>
+of a new feature must justify the additional burden on users
+to decide whether to grant permission for each feature
+whenever it's requested by a Web page.
+
+<p class="example">
+For example,
+the [Geolocation API](https://www.w3.org/TR/geolocation-API/)
+grants access to a user's location.
+This can help users in some contexts,
+like a mapping application,
+but may be dangerous to some users in other contexts -
+especially if used without the user's knowledge.
+So that the user may decide whether their location may be used by a Web page,
+a permission prompt should be shown to the user asking whether to grant location access.
+If the user refuses permission,
+no location information is available to the Web page.
+</p>
+
+<h3 id="devices-platforms">Support the full range of devices and platforms (Media Independence)</h3>
 <!-- was "Media Independence" in the HTML Design Principles -->
 
-Features on the web should work across different input and output devices,
-different platforms, and different media,
-as much as this is reasonably possible.
-For example, they should not be specific to a particular screen size,
-a particular operating system or browser engine,
-or assume the user is using a mouse.
+As much as possible,
+ensure that features on the web work across different input and output devices,
+different screen sizes, different interaction modes,
+different platforms, and different media.
```

Comment:
Feels like different could be simplified to one instance here.

Reply:
Done.

Reply:
It could, although I sometimes intentionally repeat the last common word before a branching structure in a sentence.  Doing this makes it more obvious where the sentence branches and thus avoids ambiguity (where it's not clear where the part after a comma is picking up from).